public/app/features/dashboard/services/TimeSrv.ts

```diff
@@ -65,6 +70,48 @@ export class TimeSrv {
     }
   }
 
+  checkTimeAtLoad() {
+    let maxTimeRangeInMillis = 0,
+      oldestFromInMillis = 0;
+    if (this.dashboard.timepicker.maxTimeRange) {
+      maxTimeRangeInMillis = rangeUtil.intervalToMs(this.dashboard.timepicker.maxTimeRange);
+    }
+    if (this.dashboard.timepicker.oldestFrom) {
+      oldestFromInMillis = rangeUtil.intervalToMs(this.dashboard.timepicker.oldestFrom);
+    }
+    let min = 0;
+    if (maxTimeRangeInMillis !== 0 && oldestFromInMillis !== 0) {
+      min = Math.min(maxTimeRangeInMillis, oldestFromInMillis);
+    } else if (maxTimeRangeInMillis !== 0) {
+      min = maxTimeRangeInMillis;
+    } else if (oldestFromInMillis !== 0) {
+      min = oldestFromInMillis;
+    }
+    let str;
+    if (min !== 0) {
+      str = rangeUtil.secondsToHms(min / 1000);
+    }
+    const timezone = this.dashboard ? this.dashboard.getTimezone() : undefined;
+    let from: any = this.time.from;
+    let to: any = this.time.to;
+    if (dateMath.isMathString(from)) {
+      from = dateMath.parse(from, false, timezone);
+    }
+    if (dateMath.isMathString(to)) {
+      to = dateMath.parse(to, false, timezone);
+    }
+
+    if (
+      min !== 0 &&
+      ((maxTimeRangeInMillis !== 0 && to.valueOf() - from.valueOf() > maxTimeRangeInMillis) ||
+        (oldestFromInMillis !== 0 && Date.now() - from.valueOf() > oldestFromInMillis))
+    ) {
+      this.time.from = 'now-' + str;
+      this.time.to = 'now';
```

Comment:
Good point. I can definitely see some confusion where you think you're looking at last weeks data but in reality it's this weeks data. Either we need to load no data or make it abundantly clear to the user that the time range was changed, not sure just the error message toast is enough.

Reply:
Thanks for your suggestion. I have added a warning message "Timerange of this dashboard has changed".

Reply:
We could just revert to the default time range saved in the dashboard instead? And just add an error that the time range is outside the range limit

Reply:
What if the default time range is outside the range limit? 
I know it is obvious that someone chooses the default time range within the range.
Let the default time range is 'now-6h' and the max time range limit is 3h then this could be a bug.

Reply:
I prefer if someone modifies the time range(from the URL) outside the limit then an error message should be shown(already implemented in my code) and the grafana should set the time range 'now-<max_time_limit>'